File: SelectionSort/SelectionSort3.c
#include<stdio.h>
#include<stdlib.h>

void selectionSort(int *data,int size)
{
int e,f,g,si,oep,iep;
oep=size-2;
iep=size-1;
e=0;
while(e<=oep)
{
f=e+1;
si=e;
while(f<=iep)
{
if(data[f]<data[si]) si=f;
f++;
}
g=data[si];
data[si]=data[e];
data[e]=g;
e++;
}
}

int main()
{
int *x;
int y,m,req;
printf("Enter requirement : ");
scanf("%d",&req);
fflush(stdin);
if(req<=0)
{
printf("Invalid requirement : %d",req);
return 0;
}
x=(int *)malloc(sizeof(int)*req);
if(x==NULL)
{
printf("Cannot allocate memory for the requirement : %d\n",req);
return 0;
}
m=req-1;
y=0;
while(y<=m)
{
printf("Enter a number : ");
scanf("%d",&x[y]);
fflush(stdin);
y++;
}
selectionSort(x,req);
for(y=0;y<=m;y++) printf("%d\n",x[y]);
return 0;
}

File: SelectionSort/SelectionSort3.java
import java.io.*;
class SelectionSort3
{
public static void selectionSort(int x[])
{
int size,oep,iep,e,f,g,si;
size=x.length;
oep=size-2;
iep=size-1;
for(e=0;e<=oep;e++)
{
for(si=e,f=e+1;f<=iep;f++) if(x[f]<x[si]) si=f;
g=x[e];
x[e]=x[si];
x[si]=g;
}
}
public static void main(String args[])
{
BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
int x[],req,y;
req=0;
try
{
System.out.print("Enter your requirement : ");
req=Integer.parseInt(br.readLine());
}catch(IOException ioException)
{
System.out.println(ioException);
}catch(NumberFormatException nfe)
{
System.out.println(nfe);
}
x=new int[req];
for(y=0;y<req;y++)
{
try
{
System.out.print("Enter a number : ");
x[y]=Integer.parseInt(br.readLine());
}catch(IOException ioException)
{
System.out.println(ioException);
}catch(NumberFormatException nfe)
{
System.out.println(nfe);
}
}
selectionSort(x);
for(y=0;y<req;y++) System.out.println(x[y]);
}
}
